src/main/java/com/iexec/core/task/update/TaskUpdateRequestManager.java

```diff
@@ -89,35 +109,60 @@ public void setRequestConsumer(final TaskUpdateRequestConsumer consumer) {
      */
     @Scheduled(fixedDelay = 1000)
+    @SneakyThrows
     public void consumeAndNotify() {
         if (consumer == null){
             log.warn("Waiting for consumer before consuming [queueSize:{}]", queue.size());
             return;
         }
 
-        while (!Thread.currentThread().isInterrupted()){
-            waitForTaskUpdateRequest();
+        // We should only start threads to complete the thread pool.
+        // This is useless for a fresh start but could be handy
+        // if current thread is stopped and current method is rescheduled.
+        final int nbThreadsToStart = TASK_UPDATE_THREADS_POOL_SIZE - updatingThreads.get();
+        for (int i = 0; i < nbThreadsToStart; i++) {
+            startTaskUpdateThread();
         }
     }
 
+    private void startTaskUpdateThread() {
+        // When a task update is achieved or if there's an issue,
+        // we can start another task update.
+        // This should ensure a constant number of waiting and running threads.
+        CompletableFuture.runAsync(this::waitForTaskUpdateRequest, taskUpdateExecutor)
+                .thenRunAsync(this::startTaskUpdateThread);
+    }
+
     /**
      * Wait for new task update request and run the update once a request is arrived.
      * 2 requests can be run in parallel if they don't target the same task.
      * <br>
      * Interrupts current thread if queue taking has been stopped during the execution.
      */
     private void waitForTaskUpdateRequest() {
+        updatingThreads.incrementAndGet();
         log.info("Waiting requests from publisher [queueSize:{}]", queue.size());
         try {
-            String chainTaskId = queue.take();
-            CompletableFuture.runAsync(() -> {
-                synchronized (locks.computeIfAbsent(chainTaskId, key -> new Object())){ // require one update on a same task at a time
-                    consumer.onTaskUpdateRequest(chainTaskId); // synchronously update task
-                }
-            }, taskUpdateExecutor);
+            final Task task = queue.take();
+            String chainTaskId = task.getChainTaskId();
+            log.info("Selected task [chainTaskId: {}, status: {}]", chainTaskId, task.getCurrentStatus());
+            synchronized (locks.computeIfAbsent(chainTaskId, key -> new Object())) { // require one update on a same task at a time
+                consumer.onTaskUpdateRequest(chainTaskId); // synchronously update task
+            }
         } catch (InterruptedException e) {
             log.error("The unexpected happened", e);
             Thread.currentThread().interrupt();
+        } finally {
+            updatingThreads.decrementAndGet();
         }
     }
+
+    PriorityBlockingQueue<Task> createQueue() {
+        // Tasks whose status are the more advanced should be computed before others
+        final Comparator<Task> comparator = Comparator.comparing(task -> task.getCurrentStatus().ordinal(), Comparator.reverseOrder());
```

Comment:
Why sorting them by status instead of contribution deadline?

Reply:
When we work with bag of tasks, every task has the same `contributionDeadline`... But some of them may be more advanced in their statuses, so they have priority to be updated.
I agree with you on the fact that sorting task by contribution deadline may be great and that's something that should not be too hard to implement. See https://github.com/iExecBlockchainComputing/iexec-core/pull/479#issuecomment-995699012.